src/onegov/form/display.py

```diff
@@ -121,6 +119,71 @@ def __call__(self, field: 'Field') -> Markup:
         ).format(url=field.data)
 
 
+@registry.register_for('VideoURLField')
+class VideoURLFieldRenderer(BaseRenderer):
+    """
+    Renders a video url. Embeds the video if in case of vimeo or
+    youtube otherwise just displays the url as a link.
+    """
+
+    video_template = Markup("""
+        <div class="video">
+            <div class="videowrapper">
+                <iframe allow="fullscreen" frameborder="0" src="{url}"
+                sandbox="allow-scripts allow-same-origin
+                allow-presentation" referrerpolicy="no-referrer"></iframe>
+            </div>
+        </div>
+    """)
+    url_template = Markup('<a href="{url}">{url}</a>')
+
+    def __call__(self, field: 'Field', **kwargs: Any) -> Markup:
+        url = None
+        data = self.escape(field.data)
+
+        # youtube
+        if any(x in data for x in ['youtube', 'youtu.be']):
+            url = self.ensure_youtube_embedded_url(data)
+
+        # vimeo
+        if any(x in data for x in ['vimeo']):
+            url = self.ensure_vimeo_embedded_url(data)
+
+        if url:
+            return self.video_template.format(url=self.escape(url))
+
+        # for non-vimeo and non-youtube sources we just display the url
+        return self.url_template.format(url=self.escape(data))
```

Comment:
```suggestion
        data = field.data

        # youtube
        if any(x in data for x in ['youtube', 'youtu.be']):
            url = self.ensure_youtube_embedded_url(data)

        # vimeo
        if any(x in data for x in ['vimeo']):
            url = self.ensure_vimeo_embedded_url(data)

        if url:
            return self.video_template.format(url=url)

        # for non-vimeo and non-youtube sources we just display the url
        return self.url_template.format(url=data)
```
No need to be this defensive about escaping, `Markup` already calls `escape` on any parameter of the `format` method.